### PR TITLE
BUG: fix GH45032 iloc(axis=1).__setitem__ axis argument

### LINKED ISSUE #45032: BUG: iloc(axis=1).__setitem__ ignores axis
```
import numpy as np
import pandas as pd

arr = np.random.randn(4, 4)
df = pd.DataFrame(arr)
df[4] = "foo"

df.iloc(axis=1)[2] = 3
>>> df
          0         1         2         3    4
0 -0.042240  2.894031  0.086813  0.808991  foo
1  0.141410  1.998173  0.730306 -0.779878  foo
2  3.000000  3.000000  3.000000  3.000000    3
3  0.270897  1.830372  0.533242 -0.147240  foo
```

Expected Behavior:
```
>>> df
          0         1  2         3    4
0 -0.042240  2.894031  3  0.808991  foo
1  0.141410  1.998173  3 -0.779878  foo
2  1.611364  1.459234  3  0.572484  foo
3  0.270897  1.830372  3 -0.147240  foo
```

### PR DESCRIPTION
**Pull request by ellequelle**
- [x] closes #45032 
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them
- [x] whatsnew entry

### CODE REVIEW

File: pandas/core/indexing.py [hunk 1]
```diff
@@ -1540,7 +1540,12 @@ def _convert_to_indexer(self, key, axis: int):
     def _get_setitem_indexer(self, key):
         # GH#32257 Fall through to let numpy do validation
         if is_iterator(key):
-            return list(key)
+            key = list(key)
+
+        # GH#45032 handle iloc(axis=1)
```
Comment (phofl) [inline pandas/core/indexing.py:1545]: I think you can remove that comment, this is obvious?

File: pandas/tests/indexing/test_iloc.py [hunk 2]
```diff
@@ -444,6 +444,19 @@ def test_iloc_setitem(self):
         expected = Series([0, 1, 0], index=[4, 5, 6])
         tm.assert_series_equal(s, expected)
 
+    def test_iloc_setitem_axis_argument(self):
+        # GH45032
+        dfa = DataFrame([[6, "c", 10], [7, "d", 11], [8, "e", 12]])
```
Comment (phofl) [inline pandas/tests/indexing/test_iloc.py:449]: could you do:

```
df = ...
expected = ...
df.iloc ...
tm....

df = ...
expected = ...
df.iloc...
tm...
```

Could also make two tests

**Review by phofl — CHANGES_REQUESTED** [review]
small comments

File: pandas/core/indexing.py (hunk 1 above)
Comment (ellequelle) [reply pandas/core/indexing.py:1545]: sure, should I leave the GH45032 or remove it completely?

Comment (phofl) [reply pandas/core/indexing.py:1545]: I think you can remove it completly

**Review by phofl — COMMENTED** [review]
pre-commit is failing, otherwise lgtm

**Review by phofl — APPROVED** [review]

**Comment by jreback** [comment]
thanks @ellequelle very nice!

**Pull request merged**